src/devices/Ak8963/README.md

```diff
@@ -0,0 +1,114 @@
+# AK8963 - Magnetometer
+
+The AK8963 is a magnetometer that can be controlled either thru I2C either thru SPI. It is present in other sensors like the [MPU9250](../Mpu9250/README.md). This implementation fully supports the I2C mode and the usage thru the MPU9250. It does not support SPI.
+
+## Usage
+
+You can find an example in the [sample](./samples/ak8963.sample.cs) directory. Usage is straight forward including the possibility to have a calibration.
+
+```csharp
+var mpui2CConnectionSettingmpus = new I2cConnectionSettings(1, Ak8963.Ak8963.DefaultI2cAddress);
+// This will use the default I2C interface
+Ak8963 ak8963 = new Ak8963(I2cDevice.Create(mpui2CConnectionSettingmpus));
+if (!ak8963.CheckVersion())
+    throw new IOException($"This device does not contain the correct signature 0x48 for a AK8963");
+Console.Clear();
+while (!Console.KeyAvailable)
+{
+    var magne = ak8963.ReadMagnetometer(true);
+    Console.WriteLine($"Mag X = {magne.X, 15}");
+    Console.WriteLine($"Mag Y = {magne.Y, 15}");
+    Console.WriteLine($"Mag Z = {magne.Z, 15}");
+    Thread.Sleep(100);
+}
+```
+
+## Calibration and bias
+
+You can get access to the self tests and calibration thru the ```CalibrateMagnetometer``` function which will return the bias calibration. Be aware that the calibration takes couple of seconds.
+
+```csharp
+var mag = ak8963.CalibrateMagnetometer();
+Console.WriteLine($"Bias:");
+Console.WriteLine($"Mag X = {ak8963.MagnometerBias.X}");
+Console.WriteLine($"Mag Y = {ak8963.MagnometerBias.Y}");
+Console.WriteLine($"Mag Z = {ak8963.MagnometerBias.Z}");
+```
+
+## Using a different I2C interface
```

Comment:
I think I'd prefer this part to be removed and issue about lack of the abstraction to be create but I guess I'm fine either way

Reply:
I always prefer to leave everything in the documentation. Better for anyone to remember why it's like that as well :-)

Reply:
I don't feel strongly either way, fine to leave :) I usually prefer as short docs as possible but I've seen many people prefer longer so this is a personal choice :)